### PR TITLE
Add Dell PowerEdge C6400/C6420, R340, R420, R540, R730xd, R740xd2, T640

### PR DESCRIPTION
**Pull request by rgdt**

### CODE REVIEW

File: device-types/Dell/PowerEdge_C6420.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+manufacturer: Dell
+model: PowerEdge C6420
+slug: dell_poweredge_c6420
+u_height: 2
+is_full_depth: false
+comments: |
+  [Dell C6420 Spec Sheet](https://www.delltechnologies.com/asset/en-us/products/servers/technical-support/dell-emc-poweredge-c6520-spec-sheet.pdf)
+console-server-ports:
+  - name: Rear USB
+    type: usb-a
```
Comment (DanSheps) [inline device-types/Dell/PowerEdge_C6420.yaml:11]: We are no longer modelling USB ports that are not console ports as console or console server ports

File: device-types/Dell/PowerEdge_R340.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+u_height: 1
+is_full_depth: true
+console-ports:
+  - name: Serial
+    type: de-9
+  - name: VGA
+    type: other
+console-server-ports:
+  - name: Front USB1
+    type: usb-a
```
Comment (DanSheps) [inline device-types/Dell/PowerEdge_R340.yaml:14]: This should be removed

File: device-types/Dell/PowerEdge_R540.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    type: other
+console-server-ports:
+  - name: Front USB1
+    type: usb-a
+  - name: Front USB2
+    type: usb-a
+  - name: Rear USB1
+    type: usb-a
+  - name: Rear USB2
+    type: usb-a
```
Comment (DanSheps) [inline device-types/Dell/PowerEdge_R540.yaml:22]: See previous comment about USB ports not being modelled this way

File: device-types/Dell/PowerEdge_R740xd2.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+console-ports:
+  - name: Serial
+    type: de-9
+  - name: Front VGA
+    type: other
+console-server-ports:
+  - name: Rear USB1
+    type: usb-a
+  - name: Rear USB2
+    type: usb-a
```
Comment (DanSheps) [inline device-types/Dell/PowerEdge_R740xd2.yaml:16]: See previous comment about USB ports not being modelled this way

File: device-types/Dell/PowerEdge_T640.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  - name: Rear USB2
+    type: usb-a
+  - name: Rear USB3
+    type: usb-a
+  - name: Rear USB4
+    type: usb-a
+  - name: Rear USB5
+    type: usb-a
+  - name: Rear USB6
+    type: usb-a
```
Comment (DanSheps) [inline device-types/Dell/PowerEdge_T640.yaml:28]: See previous comment about USB ports not being modelled this way

**Review by DanSheps — CHANGES_REQUESTED** [review]

**Comment by rgdt** [comment]
Fixed and applied changes requested according to your comments

**Review by DanSheps — APPROVED** [review]

**Pull request merged**